src/moonlink/src/storage/mooncake_table.rs

```diff
@@ -185,6 +188,32 @@ impl Snapshot {
     }
 }
 
+struct SnapshotStats {
+    creation_latency: Histogram,
+    registry: Registry,
+}
+
+impl SnapshotStats {
+    fn new() -> Self {
+        Self {
+            creation_latency: Histogram::new(linear_buckets(0.0, 3.0, 100)),
+            registry: Registry::default(),
+        }
+    }
+
+    pub fn register(&mut self) {
+        self.registry.register(
+            "snapshot_create_latency",
+            "Latency of snapshot creation",
+            self.creation_latency.clone(),
+        );
+    }
+
+    fn update(&mut self, time: f64) {
```

Comment:
```suggestion
    fn update_creation_latency(&mut self, time: f64) {
```
make the metrics prop explicit?